### ISSUE TITLE
Incorrect LR scheduler step count in on-policy KD training loop

### ISSUE THREAD

**Issue by xlx6** [songmzhang/KDFlow#4]
Hi, I found what looks like a correctness bug in the on-policy KD training loop.
The LR scheduler is initialized with rollout steps, but stepped per optimizer update, causing it to progress faster than intended.

<img width="1400" height="620" alt="Image" src="https://github.com/user-attachments/assets/2ca5fd7e-6153-4391-8b1e-d9b689055014" />
